pkg/ca/fileca/fileca.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package fileca
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sigstore/fulcio/pkg/ca"
+	"github.com/sigstore/fulcio/pkg/ca/x509ca"
+	"github.com/sigstore/fulcio/pkg/challenges"
+)
+
+type fileCA struct {
+	sync.RWMutex
+
+	cert *x509.Certificate
+	key  crypto.Signer
+}
+
+// NewFileCA returns a file backed certificate authority. Expects paths to a
+// certificate and key that are PEM encoded. The key must be encrypted
+// according to RFC 1423
+func NewFileCA(certPath, keyPath, keyPass string, watch bool) (ca.CertificateAuthority, error) {
+	var fca fileCA
+
+	var err error
+	fca.cert, fca.key, err = loadKeyPair(certPath, keyPath, keyPass)
+	if err != nil {
+		return nil, err
+	}
+
+	if watch {
+		watcher, err := fsnotify.NewWatcher()
```

Comment:
It might be nice to be able to `watcher.Close`, but that would require exposing it on `ca.CertificateAuthority` and doing some plumbing, so I'd open an issue to follow up, if we think it matters.